d1985r3.md

```diff
@@ -376,148 +303,114 @@ limitations we saw before `if constexpr` was introduced: Function templates
 often had to rely on helper structs to do simple things.
 
 
-### Allowing unversal template parameters in code
+### Dependent Names
 
 To make this feature easier to use we also propose that universal template
 parameters are usable in code. To make parsing possible we then have to
 disambiguate universal template parameters according to dependent name rules.
 
-Unfortunately always disambiguating as value prevents `using type =
-X<U>;` above from working as expected. U would be automatically disambiguated as
-a value at parse time and if it turns out not to be a value at instantiation
-time this is an error.
+The dependent name rules shall be modified such that dependent names passed as
+template arguments are promoted as universal template parameters. This allows
+`using type = X<U>;` above to work as expected and slighlty extends "Down with
+`typename`!".
 
-We actually have a version of this problem already with overloaded template
-functions and dependent names:
+This rule shall apply to dependent names other than universal template
+parameters, as well:
 
 ```cpp
 template<typename T> int f() { return sizeof(T); }
 template<auto V> int f() { return V; }
 
 template<typename T> int caller()
 {
-    return f<??? T::name>();
+    return f<T::name>();
 }
 ```
 
-Here the second f is called if T::name is a value, but if it is a type an
-instantiation error is caused, rather than the first f being called.
+Because `T::name` is a dependent type used as a template argument, it is
+promoted to a universal template parameter and the correct `f()` is called.
 
-With universal template parameters this problem will be more articulated,
-especially when universal template parameters are forwarded directly as in:
+This rule also applies when universal template parameters are not explicitly
```

Comment:
This is almost the same as on row 315.

Reply:
Yes, Line 315 was a summary saying that this rule change affects UTPs and dependent names in general. This line goes into detail on the second half, extending "Down with `typename`!"

Reply:
And now that change is gone? Am I seeing the wrong diff?